Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionFragmentTest.kt

```diff
@@ -0,0 +1,128 @@
+import de.rki.coronawarnapp.dccticketing.ui.certificateselection.cards.DccTicketingNoValidCertificateFaqCard
+import de.rki.coronawarnapp.dccticketing.ui.certificateselection.cards.DccTicketingNoValidCertificateHeaderCard
+import de.rki.coronawarnapp.dccticketing.ui.certificateselection.cards.DccTicketingVaccinationCard
+import de.rki.coronawarnapp.dccticketing.ui.certificateselection.cards.DccTicketingValidCertificateHeaderCard
+import de.rki.coronawarnapp.dccticketing.ui.shared.DccTicketingSharedViewModel
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import org.joda.time.LocalDate
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseUITest
+import testhelpers.Screenshot
+import testhelpers.launchFragmentInContainer2
+import testhelpers.takeScreenshot
+
+@RunWith(AndroidJUnit4::class)
+class DccTicketingCertificateSelectionFragmentTest : BaseUITest() {
+
+    @MockK lateinit var viewModel: DccTicketingCertificateSelectionViewModel
+    private val vc = DccTicketingValidationCondition(
+        lang = "en-en",
+        fnt = "SCHNEIDER",
+        gnt = "ANDREA",
+        dob = "1989-12-12",
+        coa = "AF",
+        cod = "SJ",
+        roa = "AF",
+        rod = "SJ",
+        type = listOf(
+            "r",
+            "v",
+            "t"
+        ),
+        category = listOf(
+            "Standard"
+        ),
+        validationClock = "2021-11-03T15:39:43+00:00",
+        validFrom = "2021-11-03T07:15:43+00:00",
+        validTo = "2021-11-03T15:39:43+00:00",
+        hash = null
+    )
+
+    private val navController = TestNavHostController(
+        ApplicationProvider.getApplicationContext()
+    ).apply {
+        UiThreadStatement.runOnUiThread {
+            setViewModelStore(ViewModelStore())
+            setGraph(R.navigation.nav_graph)
+            setCurrentDestination(R.id.dccTicketingCertificateSelectionFragment)
+        }
+    }
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxed = true)
+        setupMockViewModel(
+            object : DccTicketingCertificateSelectionViewModel.Factory {
+                override fun create(
+                    dccTicketingSharedViewModel: DccTicketingSharedViewModel
+                ): DccTicketingCertificateSelectionViewModel = viewModel
+            }
+        )
+    }
+
+    @Screenshot
+    @Test
+    fun dccTicketingCertificateSelectionFragment_no_certificates() {
+        every { viewModel.items } returns MutableLiveData(
+            listOf(
+                DccTicketingNoValidCertificateHeaderCard.Item(),
+                DccTicketingNoValidCertificateCard.Item(vc),
+                DccTicketingNoValidCertificateFaqCard.Item()
+            )
+
+        )
+        launchFragmentInContainer2<DccTicketingCertificateSelectionFragment>(
+            testNavHostController = navController
+        )
```

Comment:
@AlexanderAlferov  you can pass `testNavController`   as https://github.com/corona-warn-app/cwa-app-android/pull/4439/files#diff-e4121ff9898e0ed0575c9dd9ae6e547812252047c5251864ca97eceadcb52cc9R61-R69
with the current destination ofc